go/libkb/sig_chain.go

```diff
@@ -370,6 +381,29 @@ func (sc *SigChain) VerifyChain(m MetaContext) (err error) {
 		curr.chainVerified = true
 	}
 
+	newHPrevInfo := NewInitialHPrevInfo()
+	for i := 0; i < len(sc.chainLinks); i++ {
```

Comment:
This will be a major performance regression.  Looking at the full sigchain loading code. note that `VerifyChain` is called several times, but short-circuits due to `curr.chainVerified`.  Your code is just going to plow over the whole chain repeatedly.  I think it needs to be better.

Reply:
You can use the same structure as what's already there to solve this issue btw. I would recommend attaching an hprev to those links that don't supply it explicitly, so you can pick up traversal where you left off.

And *this* is why we go backwards.  I think you need to go backwards too.

Reply:
Ok, simple fix!  Instead of `i := 0`, just do `i := <when the above loop stopped>`

Sorry for the churn...